cores/arduino/stm32/analog.c

```diff
@@ -110,11 +112,52 @@
   * @}
   */
 
-/** @addtogroup STM32F4xx_System_Private_Variables
-  * @{
-  */
+/* Adc context                 */
+AdcContext context[ADC_NUM];
+
+/* DMA is use, or classic ADC  */
+static bool use_dma = false;
```

Comment:
I still don"t understand why use_dma is not defined for each instance ... how does it work in case ADC1 is used in DMA mode wile ADC2 is used without DMA.

Reply:
Ok right, i see where is the problem. My thing works only for initialization, but I have lost the use_dma information when I wat Deinit it. So yes it's better to make a use_dma for each instance.